dev/AppNotifications/AppNotificationActivatedEventArgs.cpp

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "AppNotificationActivatedEventArgs.h"
+#include "Microsoft.Windows.AppNotifications.AppNotificationActivatedEventArgs.g.cpp"
+#include "AppNotificationBuilder/AppNotificationBuilderUtility.h"
+
+namespace winrt::Microsoft::Windows::AppNotifications::implementation
+{
+    winrt::Windows::Foundation::Collections::IMap<hstring, hstring> AppNotificationActivatedEventArgs::DecodeArguments(std::wstring arguments)
+    {
+        auto result{ winrt::single_threaded_map<winrt::hstring, winrt::hstring>() };
+
+        std::vector<std::wstring> pairs{};
+        size_t pos{ 0 };
+
+        // Separate the key/value pairs by ';' as the delimiter
+        while ((pos = arguments.find(L';')) != std::wstring::npos)
```

Comment:
This is a much easier way to tokenize:

int main()
{
    char str[] = "Geeks-for-Geeks";
 
    // Returns first token
    char *token = strtok(str, " ");
 
    // Keep printing tokens while one of the
    // delimiters present in str[].
    while (token != NULL)
    {
        printf("%s\n", token);
        token = strtok(NULL, " ");
    }
 
    return 0;

Reply:
This would avoid that last push_back that you have outside the while loop

Reply:
And you don't have to do that erase :)

Reply:
Yes, this can be improved. We'll address this in the feature level nits PR that will go in later today.